DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -88,22 +88,37 @@ public IActionResult EditDetails(EditDetailsViewModel model, string action)
 
         private IActionResult EditDetailsPostSave(EditDetailsViewModel model)
         {
-            ValidateCustomPrompts(model);
+            var userAdminId = User.GetAdminId();
+            var userDelegateId = User.GetNullableCandidateId();
 
-            if (!ModelState.IsValid)
+            if (userDelegateId.HasValue)
             {
-                return View(model);
+                ValidateJobGroup(model);
+                ValidateCustomPrompts(model);
             }
 
             if (model.ProfileImageFile != null)
             {
                 ModelState.AddModelError(nameof(EditDetailsViewModel.ProfileImageFile),
                     "Preview your new profile picture before saving");
-                return View(model);
             }
 
-            var userAdminId = User.GetAdminId();
-            var userDelegateId = User.GetNullableCandidateId();
+            var (adminUser, delegateUser) = userService.GetUsersById(userAdminId, userDelegateId);

```

Comment:
I feel like it might be worth putting this logic into a service and writing a few unit tests for it. Then, the controller will be simplified and will only need something like:
```
if (!NewEmailAddressIsValid(model.Email!, adminUserId, delegateUserId) {
   ModelState.AddModelError(...)
}
```